GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -65,9 +68,37 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
             generateNotifications();
         } else if ("purgeDeviceFileJobQueueRecords".equals(action)) {
             purgeDeviceFileJobQueueRecords();
+        } else if ("purgeReportRecords".equals(action)) {
+            purgeReportRecords();
         }
     }
 
+    /**
+     * scans for and deletes Report entries that are either more than one year old
+     */
+    private void purgeReportRecords() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.YEAR, ONE_YEAR_AGO);
+        log.info("Starting scan for Report entries, older than: " + deadline.getTime());
+        ReportDao reportDao = new ReportDao();
+        List<Report> reportList = reportDao.list("all"); //TODO: only list old entities
+        int retirees = 0;
+        for (Report item : reportList) {
+            if (item.getCreatedDateTime() != null
+                    && deadline.getTime().after(item.getCreatedDateTime())) {
+                //cheap case - old
+                log.info("Deleting old Report entry: " + item.getKey().getId());
+                //TODO delete file from file store?
+                //If that succeeded:
+                SurveyTaskUtil.spawnDeleteTask(SurveyTaskRequest.DELETE_REPORT_ACTION,
```

Comment:
We have the function `BaseDAO.delete(Collection)` that can delete all reports retrieved in one shot.

Reply:
I thought we needed to make sure it does not hit the 60-second limit.

Reply:
I think sending a single delete should not be a problem for not hitting the 60 second limit. I don't think we will be talking about more than 1,000 reports per week for an entire instance.